k8srm-prototype/pkg/api/capacity_types.go

```diff
@@ -53,6 +61,58 @@ type DevicePoolStatus struct {
 	AvailableDevices int `json:"availableDevices,omitempty"`
 }
 
+// Device is used to track individual devices in a pool.
+type Device struct {
+	// Name is a driver-specific identifier for the device.
+	// +required
+	Name string `json:"name"`
+
+	// Attributes contain additional metadata that can be used in
+	// constraints. If an attribute name overlaps with the pool attribute,
+	// the device attribute takes precedence.
+	//
+	// +optional
+	Attributes []Attribute `json:"attributes,omitempty"`
+
+	// Requests contains the pooled resources that are consumed when
+	// this device is allocated.
+	//
+	// +optional
+	Requests map[string]resource.Quantity `json:"requests,omitempty"`
+
+	// Resources allows the definition of per-device resources that can
+	// be allocated in a manner similar to standard Kubernetes resources.
+	// NOTE: We may not need to implement this right away.
+	//
+	// +optional
+	Resources []ResourceCapacity `json:"resources,omitempty"`
+}
+
+type ResourceCapacity struct {
+	// Name is the resource name/type.
+	Name string `json:"name"`
+
+	// Capacity is the total capacity of the named resource.
+	// +required
+	Capacity resource.Quantity `json:"capacity"`
+
+	// BlockSize is the increments in which capacity is consumed. For
+	// example, if you can only allocate memory in 4k pages, then the
+	// block size should be "4Ki". Default is 1.
```

Comment:
The problem is, it's up to the device what is "natural" and the scheduler and device have to agree on that, so we need it communicated from the device.

Reply:
I guess I am assuming we are getting away with it for CPUs right now because they are similar enough. We probably have some default quantum for memory and CPU that's good enough for everyone. I don't know if that's true across all these devices.

Reply:
I've left this in for now...

Reply:
OK to resolve this